nca/NetworkConfig/NetworkLayer.py

```diff
@@ -252,6 +253,17 @@ def collect_policies_conns_optimized(self, is_ingress, captured_func=lambda poli
         """
         res_conns = OptimizedPolicyConnections()
         for policy in self.policies_list:
+            # Track the peers that were affected by this policy
+            if ExplTracker().is_active():
+                for peer in policy.selected_peers:
+                    src_peers, _ = ExplTracker().extract_peers(policy.optimized_allow_ingress_props)
+                    _, dst_peers = ExplTracker().extract_peers(policy.optimized_allow_egress_props)
```

Comment:
What about `policy.optimized_deny_ingress_props` and `policy.optimized_deny_egress_props`? They may also explain a lack of connections between certain peers

Reply:
In the case of lack of connectivity we show all the policies that affects the queried peers, since also a typo in a 'non deny' policy may cause the lack of connectivity.